src/storage/v2/inmemory/storage.cpp

```diff
@@ -1039,8 +1039,30 @@ utils::BasicResult<StorageManipulationError, void> InMemoryStorage::InMemoryAcce
 
 // NOLINTNEXTLINE(google-default-arguments)
 utils::BasicResult<StorageManipulationError, void> InMemoryStorage::InMemoryAccessor::PeriodicCommit(
-    CommitReplArgs /*reparg*/, DatabaseAccessProtector /*db_acc*/) {
-  throw utils::NotYetImplemented("Periodic commit is not yet implemented on in-memory modes.");
+    CommitReplArgs reparg, DatabaseAccessProtector db_acc) {
+  auto result = Commit(reparg, db_acc);
+
+  if (result.HasError()) {
+    return result;
+  }
+
+  FinalizeTransaction();
+
+  auto original_start_timestamp = !transaction_.original_start_timestamp.has_value()
+                                      ? transaction_.start_timestamp
+                                      : transaction_.original_start_timestamp.value();
+
+  auto *mem_storage = static_cast<InMemoryStorage *>(storage_);
+
+  auto new_transaction = mem_storage->CreateTransaction(transaction_.isolation_level, transaction_.storage_mode);
```

Comment:
Might need a way to kill a periodic commit transaction using a global uuid.

Reply:
@andrejtonev this should be fine, we already decoupled interpreter txn id from the storage txn id.

Reply:
I have tested this. Interpreter still holds the initial transaction ID and that one is used to kill the periodic transaction successfully.
The transaction ID's that are subsequent in committing don't affect this flow.